rich/console.py

```diff
@@ -656,8 +656,12 @@ def __init__(
 
         self.is_jupyter = _is_jupyter() if force_jupyter is None else force_jupyter
         if self.is_jupyter:
-            width = width or 93
-            height = height or 100
+            jupyter_columns = self._environ.get("JUPYTER_COLUMNS", "")
```

Comment:
N.B. this works the same way, but is written slightly differently, from how `COLUMNS`/`LINES` are handled in a couple of other places in the file, e.g.
```
columns = self._environ.get("COLUMNS")
        if columns is not None and columns.isdigit():
            width = int(columns)
```
The reason for this was to not overcomplicate the `if` with another conditional, given it's already following an `or`. I could change the other `COLUMNS`/`LINES` parts in this file to match what I've done here or vice versa.

Reply:
We're looking up `JUPYTER_COLUMNS` every time, even when `width` is non zero. 

Suggest structuring it so that it is clear we only use it when `width` is not set. i.e.

`if not width:`